experimental/ConsoleApp6/ConsoleApp6/Api.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+namespace System.Diagnostics
+{
+    public class MetricSource
+    {
+        public MetricSource(string componentName, Version version = default, IDictionary<string,string> defaultLabels = null) { }
+        
+        public Counter CreateCounter(string counterName, IDictionary<string,string> defaultLabels = null)
+        {
+            Metric m = new Metric(this, counterName, MetricAggregations.Sum, defaultLabels);
+            return new Counter(m);
+        }
+
+        public Gauge CreateGauge(string counterName, IDictionary<string, string> defaultLabels = null)
+        {
+            Metric m = new Metric(this, counterName, MetricAggregations.Sum, defaultLabels);
+            return new Gauge(m);
+        }
+    }
+
+
+    //TODO: do we have different metric types or a generic parameter to allow collection of different integer/float sizes?
+    public class Metric
+    {
+        internal Metric(MetricSource source,
+                        string name,
+                        MetricAggregation defaultAggregation,
+                        IDictionary<string, string> defaultLabels)
+        {
+            Source = source;
+            Name = name;
+            DefaultAggregation = defaultAggregation;
+            DefaultLabels = new LabelSet(defaultLabels);
+        }
+
+        public MetricSource Source { get; }
+        public string Name { get; }
+        public MetricAggregation DefaultAggregation { get; }
+        public LabelSet DefaultLabels { get; }
+
+        public void Record(int measurement) { }
+        public void Record(int measurement, string labelName1, string labelValue1) { }
+        public void Record(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2) { }
+        public void Record(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3) { }
+        public void Record(int measurement, string labelName1, string labelValue1, string labelName2, string labelValue2, string labelName3, string labelValue3, 
+            params string[] additionalLabels) { }
+        public void Record(int measurement, LabelSet labels)
+        { }
+
+    }
+
+    //TODO: we could derive from Metric but that would expose the Record() API. Does that matter?
+    //TODO: If we don't derive from Metric perhaps we should replicate the other properties?
+    public struct Counter
```

Comment:
I noticed you have a struct for Counter as a wrapper for a Metric class.  Is this faster?

Reply:
It was a silly way to supress the APIs that would have shown up if I had inherited. What I should have done (and at this point have done) is mark the APIs protected.